docs/source/plugin/tutorials/playing-with-commands.rst

```diff
@@ -0,0 +1,152 @@
+    <Sopel> Hello YourNick, have a nice day!
+
+For that, you need **the second argument**: the ``trigger``. It is an object
+with information about the message that triggered your
+callable, such as the **message** itself, the **channel**, the type of message,
+etc.—and what we need for now is the
+:attr:`trigger.nick <sopel.trigger.Trigger.nick>` attribute::
+
+    from sopel import plugin
+
+    @plugin.command('hello')
+    def hello(bot, trigger):
+        """Say Hello <user>, have a nice day!"""
+        bot.say(f'Hello {trigger.nick}, have a nice day!')
+
+.. important::
+
+    If you want to test this with your bot, and your bot is already running,
+    restart the bot so it will load the new version of your plugin.
+
+.. seealso::
+
+    You can learn much more about the :class:`trigger <sopel.trigger.Trigger>`
+    object by reading its documentation.
+
+
+Command with arguments
+======================
+
+The trigger object can do much more for you: if a user adds arguments to the
+command, like ``.hello morning``, you can detect and use that argument::
+
+    from sopel import plugin
+
+    @plugin.command('hello')
+    def hello(bot, trigger):
+        """Say Hello <user>, have a nice day!"""
+        # group 1 is the name of the command that was triggered
+        # group 2 is the entire rest of the message
+        # groups 3 to 6 are the first, second, third, and fourth command arg
+        when = trigger.group(3)
+        # select a different greeting depending on when
+        greeting = {
+            'morning': 'and good morning!',
+            'noon': 'are you having lunch?',
+            'night': 'I hope it was a good day!',
+            'evening': 'good evening to you!'
+        }.get(when, 'have a nice day!')  # default to "nice day"
+        # say hello
+        bot.say(f'Hello {trigger.nick}, {greeting}')
+
+Now the command will be able to react a bit more to your user::
+
+    <YourNick> .hello morning
+    <Sopel> Hello YourNick, and good morning!
+    <YourNick> .hello noon
+    <Sopel> Hello YourNick, are you having lunch?
+
+How does that work? Well, the short version is that Sopel uses regex
+(`REGular EXpressions`__) to match a message to a plugin callable, and the
+``trigger`` object exposes the match result.
+
+.. seealso::
+
+    You can learn much more about the :class:`~sopel.plugin.command` decorator
+    by reading its documentation.
+
+.. note::
+
+    In the case of a command, the regex is entirely managed by Sopel itself,
+    while the generic :func:`@plugin.rule <sopel.plugin.rule>` decorator
+    allows you to define your own regex.
+
+.. __: https://en.wikipedia.org/wiki/Regular_expression
+
+
+And... action!
+==============
+
+Some users say ``.hello`` out loud, and others will say it with an action. How
+do you react to these? Let's go back to the example of the ``shrug`` command:
+it uses a second decorator, ``action_command('shrugs')``, with a different
+name. How does that work?
```

Comment:
At first I thought, "We should link back to the example above."

Then I realized there would be no clear way for the reader to scroll back down. It wouldn't be ideal for navigation. But the example is so short, we might as well just repeat it for clarity.

```suggestion
do you react to these? Let's go back to the example of the ``shrug`` command::

    @plugin.command('shrug')
    @plugin.action_command('shrugs')
    def shrug(bot, trigger):
        bot.say('¯\\_(ツ)_/¯')

Notice that it also uses a second decorator, ``action_command('shrugs')``,
with a different name. How does that work?
```